isyfact-standards-doc/src/docs/antora/modules/referenzarchitektur/pages/software-technisch/backend/anwendungskern.adoc

```diff
@@ -0,0 +1,201 @@
+
+[[namenskonvention-geschaeftsobjekt]]
+.Namenskonvention Geschäftsobjekt
+[cols="1s,4m",options="header"]
+|====
+2+|Geschäftsobjekt
+|Schema |<Entitaetsname>Bo
+|Beispiel |AkteBo
+|====
+
+// end::namenskonvention[]
+
+In einer allgemeinen Anwendung gibt es z.B. die Geschäftsobjekte `Aufgabe`, `Zeitpunkt` und `Nutzer`.
+
+Eine *Anwendungsfallklasse* setzt genau einen Anwendungsfall um.
+Ihr Schnitt ergibt sich aus der fachlichen Architektur.
+Anwendungsfallklassen werden anhand des Namens des Anwendungsfalls aus der Spezifikation benannt und mit dem Präfix `Awf` gekennzeichnet.
+
+// tag::namenskonvention[]
+
+[[namenskonvention-anwendungsfallklassen]]
+.Namenskonvention Anwendungsfallklassen
+[cols="1s,4m",options="header"]
+|====
+2+|Anwendungsfallklassen
+|Schema |Awf<Anwendungsfall>
+|Beispiele |AwfAntragVerarbeiten
+
+AwfEntscheidungDurchfuehren
+|====
+
+// end::namenskonvention[]
+
+Eine *Anwendungsfunktion* setzt Teil-Abläufe um, die in mehreren Anwendungsfällen verwendet werden.
+Anwendungsfunktionen werden somit von Anwendungsfallklassen aufgerufen.
+Anwendungsfunktionen werden anhand des Namen der Anwendungsfunktion aus der Spezifikation benannt und mit dem Präfix `Afu` gekennzeichnet.
+
+// tag::namenskonvention[]
+
+[[namenskonvention-anwendungsfunktion]]
+.Namenskonvention Anwendungsfunktion
+[cols="1s,4m",options="header"]
+|====
+2+|Anwendungsfunktion
+|Schema |Afu<Anwendungsfunktion>
+|Beispiele |AfuBerechnungFristdatum
+
+AfuErmittlungEntscheidungsrelevanz
+|====
+
+// end::namenskonvention[]
+
+Die folgende Abbildung fasst den <<image-aufbau-fachkomponente>> noch einmal grafisch zusammen.
+
+[[image-aufbau-fachkomponente]]
+.Aufbau einer Fachkomponente
+image::software-technisch/backend/aufbau_fachkomponente_awk.dn.svg[]
+
+[[das-anwendungskern-framework]]
+== Anwendungskern-Framework
+
+// tag::awk-framework-definition[]
+
+Für querschnittliche Funktionalität innerhalb des Anwendungskerns wird das Spring-Framework genutzt.
+Hauptaufgabe des Frameworks ist es, die Komponenten zu konfigurieren und miteinander bekanntzumachen.
+Dadurch wird die Trennung zwischen Fachlichkeit und Technik verbessert.
+Beispiel für querschnittliche Funktionalität ist die deklarative Steuerung von Transaktionen.
+
+// end::awk-framework-definition[]
+
+Allgemeine Vorgaben hierzu sind in der xref:software-technisch/backend/spring.adoc[] beschrieben.
+
+[[konfiguration-fachkomponente]]
+=== Konfiguration von Fachkomponenten
+
+Fachkomponenten werden durchgängig mittels Spring konfiguriert.
+Das umfasst vor allem, aber nicht ausschließlich, die Fassade (Schnittstellen der Fachkomponente) und zugehörige Anwendungsfallklassen.
+
+Bei Anwendungsfunktionsklassen oder Hilfsklassen ist je nach Relevanz zu entscheiden, ob diese als eigene Spring Beans definiert werden.
+Im Zweifel sollte die Konfiguration über Spring bevorzugt werden.
+Wenn eine Klasse nur an einer Stelle genutzt wird, kann sie als Kompromiss auch als anonyme Spring Bean definiert werden.
```

Comment:
den Einschub "als Kompromiss" könnte man hier streichen

Reply:
Für dieses Ticket out of scope. Ich nehme das Thema mit.